ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,493 @@
+                hduhdr[key] = (bnch.value, bnch.comment)
+
+            # Update existing keyword
+            elif hduhdr[key] != bnch.value:
+                hduhdr[key] = bnch.value
+
+    def _write_mosaic(self, key, outfile):
+        """Write out mosaic data (or any new data generated within Ginga)
+        to single-extension FITS.
+
+        """
+        maxsize = self.settings.get('max_mosaic_size', 1e8)  # Default 10k x 10k
+        channel = self.fv.get_channelInfo()
+        image = channel.datasrc[key]
+
+        # Prevent writing very large mosaic
+        if (image.width * image.height) > maxsize:
+            s = 'Mosaic too large to be written {0}'.format(image.shape)
+            self.w.status.set_text(s)
+            self.logger.error(s)
+            return
+
+        # Insert mosaic data and header into output HDU
+        hdu = fits.PrimaryHDU(image.get_data())
+        self._write_header(image, hdu)
+
+        # Write history to PRIMARY
+        self._write_history(key, hdu)
+
+        # Write to file
+        hdu.writeto(outfile, clobber=True)
+
+    def _write_mef(self, key, extlist, outfile):
+        """Write out regular multi-extension FITS data."""
+        channel = self.fv.get_channelInfo()
+        with fits.open(outfile, mode='update') as pf:
+            # Process each modified data extension
+            for idx in extlist:
+                k = '{0}[{1}]'.format(key, self._format_extname(idx))
+                image = channel.datasrc[k]
+
+                # Insert data and header into output HDU
+                pf[idx].data = image.get_data()
+                self._write_header(image, pf[idx])
+
+            # Write history to PRIMARY
+            self._write_history(key, pf['PRIMARY'])
+
+    def toggle_save_cb(self, w, res_dict):
+        """Only enable saving if something is selected."""
+        if len(res_dict) > 0:
+            self.w.save.set_enabled(True)
+        else:
+            self.w.save.set_enabled(False)
+        return
+
+    def save_images(self):
+        """Save selected images.
+
+        This uses Astropy FITS package to save the outputs no matter
+        what user chose to load the images.
+
+        """
+        res_dict = self.treeview.get_selected()
+        clobber = self.settings.get('clobber', False)
+        self.treeview.clear_selection()  # Automatically disables Save button
+
+        # If user gives empty string, no suffix.
+        if self.suffix:
+            sfx = '_' + self.suffix
+        else:
+            sfx = ''
+
+        # Process each selected file. Each can have multiple edited extensions.
+        for infile in res_dict:
+            f_pfx = os.path.splitext(infile)[0]  # prefix
+            f_ext = '.fits'  # Only FITS supported
+            oname = f_pfx + sfx + f_ext
+            outfile = os.path.join(self.outdir, oname)
+
+            s = 'Writing out {0} to {1} ...'.format(infile, oname)
+            self.w.status.set_text(s)
```

Comment:
This can stretch out the plugin widget's width if filename is very long (which is probably true for all JWST images). Should we consider using `ginga.util.iohelper.shorten_name()` here for the informational text?